grove/ising/ising_qaoa.py

```diff
@@ -38,41 +46,50 @@ def print_fun(x):
 
 
 def ising_trans(x):
-    # Transformation to Ising notation
+    """
+    Transformation to Ising notation.
+
+    :param x: (int) Value of a single binary bit from {0, 1}.
+    :return: Transformed bit value from {-1, 1}.
+    :rtype: Integer.
+    """
     if x == 1:
         return -1
     else:
         return 1
 
 
-def ising(h, J, num_steps=0, verbose=True, rand_seed=None, connection=None, samples=None,
-          initial_beta=None, initial_gamma=None, minimizer_kwargs=None,
-          vqe_option=None):
+def ising_qaoa(h, J, num_steps=0, driver_operators=None, verbose=True,
+               rand_seed=None, connection=None, samples=None,
+               initial_beta=None, initial_gamma=None, minimizer_kwargs=None,
+               vqe_option=None):
     """
-    Ising set up method
+    Ising set up method for QAOA. Supports 2-local as well as k-local interaction terms.
 
-    :param h: External magnectic term of the Ising problem. List.
-    :param J: Interaction term of the Ising problem. Dictionary.
+    :param h: (dict) External magnectic term of the Ising problem.
+    :param J: (dict) Interaction terms of the Ising problem (may be k-local).
     :param num_steps: (Optional.Default=2 * len(h)) Trotterization order for the
                   QAOA algorithm.
+    :param driver_operators: (Optional. Default: X on all qubits.) The mixer/driver
+                Hamiltonian used in QAOA. Can be used to enforce hard constraints
+                and ensure that solution stays in feasible subspace.
+                Must be PauliSum objects.
     :param verbose: (Optional.Default=True) Verbosity of the code.
     :param rand_seed: (Optional. Default=None) random seed when beta and
                       gamma angles are not provided.
     :param connection: (Optional) connection to the QVM. Default is None.
     :param samples: (Optional. Default=None) VQE option. Number of samples
                     (circuit preparation and measurement) to use in operator
-                    averaging.
+                    averaging. Required when using QPU backend.
```

Comment:
Is this only required for the QPU? Don't the QVM and QPU both have default `num_trials`?